### PR TITLE
Updates for Aff 4.0.0

### PR DESCRIPTION
**Pull request by dgendill**
I've made some changes to support the new 4.0.0 release of Aff. Before we push this, we'll need to wait for 
[this one](https://github.com/bodil/purescript-test-unit/pull/37) in purescript-test-unit to be merged.

### CODE REVIEW

**Comment by dgendill** [comment]
Ok, I think this should be ready to push now. @Risto-Stevcev does this all look good to you?

#### v3.0.0 Changes:
- Support for Aff 4.0.0
- Removed Charset type alias. Now using Data.TextEncoder.Encoding sum type.
- Changed functions in Page and Stream to return Unit so you don't have to discard results when using do-notation
- Added Stream.seek, Stream.read, and tests for Streams.
- Updated examples

File: examples/Example.purs [hunk 1]
```diff
@@ -1,16 +1,18 @@
...
 import Control.Monad.Eff.Console (CONSOLE)
+import Control.Monad.Eff.Console as Effc
+import Control.Monad.Eff.Exception (message, stack)
+import Data.Either (Either(..))
+import Data.Maybe (fromMaybe)
+import Data.Tuple (Tuple(..))
 import PhantomJS.Page (open, render, createPage, injectJs, evaluate, Page, customHeadersRaw, png)
 import PhantomJS.Phantom (PHANTOMJS, exit)
-import Data.Tuple (Tuple(..))
+import Prelude (Unit, bind, ($), (<>), pure, discard)
```
Comment (Risto-Stevcev) [inline examples/Example.purs]: I found out that you can have one module that's open without purescript issuing a warning about it (which is mostly used for Prelude). You can keep Prelude open, ie just: `import Prelude`, without listing all the imports

File: test/Main.purs [hunk 2]
```diff
@@ -1,35 +1,46 @@
...
           )
           Unit
-main = do
+main = runAff_ (case _ of
+    Left err -> do
+      log $ "ERROR: " <> message err
+      log $ show (stack err)
+    Right r -> log "Success"
+  ) $ do
+  b <- makeEmptyVar
```
Comment (Risto-Stevcev) [inline test/Main.purs]: `b` isn't used

File: test/Main.purs [hunk 3]
```diff
@@ -1,35 +1,46 @@
 module Test.Main where
 
-import Control.Monad.Aff (Aff)
-import Control.Monad.Aff.AVar (AVAR)
+import Control.Monad.Aff (message, runAff_)
+import Control.Monad.Aff.AVar (AVAR, makeEmptyVar)
```
Comment (Risto-Stevcev) [inline test/Main.purs]: `makeEmptyVar` isn't used

File: src/PhantomJS/Page.js [hunk 4]
```diff
@@ -8,23 +12,56 @@ function PhantomPageError(message, stack) {
...
 
-exports.createPage_ = function(success, error) {
+var phantomGlobal = {
+  pageCounter : 0,
+  errors : {}
+};
+
+exports.createPage_ = function(error, success) {
   var webpage = require('webpage').create();
+  webpage.phantomUniqueId = phantomGlobal.pageCounter++;
```
Comment (Risto-Stevcev) [inline src/PhantomJS/Page.js]: Get rid of unused line

File: src/PhantomJS/Page.js [hunk 5]
```diff
@@ -8,23 +12,56 @@ function PhantomPageError(message, stack) {
 PhantomPageError.prototype = Object.create(Error.prototype);
 PhantomPageError.prototype.constructor = PhantomPageError;
 
-exports.createPage_ = function(success, error) {
+var phantomGlobal = {
```
Comment (Risto-Stevcev) [inline src/PhantomJS/Page.js]: Get rid of unused phantomGlobal

**Comment by dgendill** [comment]
Thanks for the review! I've pushed out some code that I think covers your comments.

If you test this locally, you may need re-run bower install. The `makeEmptyVar` in Test.Main was a workaround for an issue in purescript-avar.  Now that that's been fixed, you may need to get that updated dependency for the tests to work.

**Comment by Risto-Stevcev** [comment]
This is awesome, thanks for the PR!

**Pull request merged**